src/components/RadioButtonLabelNew/RadioButtonLabelNew.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { FC, ReactNode } from 'react'
+import styled, { css } from 'styled-components'
+
+import { Theme, useTheme } from '../../hooks/useTheme'
+import { useId } from '../../hooks/useId'
+import { useClassNames } from './useClassNames'
+
+import { RadioButton, Props as RadioButtonProps } from '../RadioButton'
+
+type Props = RadioButtonProps & {
+  children?: ReactNode
+}
+
+export const RadioButtonLabelNew: FC<Props> = ({ children, className = '', ...props }) => {
+  const theme = useTheme()
+  const classNames = useClassNames()
+  const radioButtonId = useId()
```

Comment:
props から id が渡ってくる可能性があるため、 `useId()` の引数として初期値を渡したほうが良さそうです。

```suggestion
  const radioButtonId = useId(props.id)
```